### PR TITLE
Add grouped conv bwd weight multi d kernel

### PR DESCRIPTION
**Pull request by bartekxk**

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp [hunk 1]
```diff
@@ -0,0 +1,1106 @@
...
+    ignore = a_element_op;
+    ignore = b_element_op;
+    ignore = c_element_op;
+    ignore = batch_count;
+    ignore = block_2_ctile_map;
+    ignore = compute_ptr_offset_of_batch;
+
+    compute_ptr_offset_of_batch.GetAPtrOffset(0);
+    compute_ptr_offset_of_batch.GetBPtrOffset(0);
+    compute_ptr_offset_of_batch.GetCPtrOffset(0);
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp:105]: Why do you need those three calls ?

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp [hunk 2]
```diff
@@ -0,0 +1,1106 @@
...
+        is_same_v<OutLayout, tensor_layout::convolution::GNHWK>;
+    // 3d
+    static constexpr bool is_NDHWGK_GKZYXC_NDHWGC =
+        is_same_v<InLayout, tensor_layout::convolution::NDHWGC> &&
+        is_same_v<WeiLayout, tensor_layout::convolution::GKZYXC> &&
+        is_same_v<OutLayout, tensor_layout::convolution::NDHWGK>;
+    static constexpr bool is_GNDHWK_GKZYXC_GNDHWC =
+        is_same_v<InLayout, tensor_layout::convolution::GNDHWC> &&
+        is_same_v<WeiLayout, tensor_layout::convolution::GKZYXC> &&
+        is_same_v<OutLayout, tensor_layout::convolution::GNDHWK>;
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp]: You could put this code into tensor layout file. In create a constexpr function like :
```
template <InLayout, WeiLayout, OutLayout>
bool is GNWK_GKXC_GNWC( in_lay, wei_lay, out_lay) { 
return is_same_v<InLayout, tensor_layout::convolution::GNWC> &&
        is_same_v<WeiLayout, tensor_layout::convolution::GKXC> &&
        is_same_v<OutLayout, tensor_layout::convolution::GNWK>;
}
```

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp [hunk 3]
```diff
@@ -0,0 +1,1106 @@
...
+                             device::ConvolutionBackwardWeightSpecialization::Filter1x1Stride1Pad0)
+                {
+                    return wei_grid_desc;
+                }
+                else
+                {
+                    const index_t GemmM = K;
+                    const index_t GemmN = C * X;
+                    const auto PadGemmM = (MPerBlock - GemmM % MPerBlock) % MPerBlock;
+                    const auto PadGemmN = (NPerBlock - GemmN % NPerBlock) % NPerBlock;
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp]: Isn't this last modulo redundant ? This first part within braces is already less than M[N]Perblock.

**Review by aosewski — DISMISSED** [review]
Reviewed device op impl - looks good. Great work as usual!

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp (hunk 3 above)
Comment (bartekxk) [reply include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp]: You're right

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp (hunk 2 above)
Comment (bartekxk) [reply include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp]: Moved to grouped_conv_utils.hpp

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp (hunk 1 above)
Comment (bartekxk) [reply include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_multiple_d_xdl_cshuffle.hpp:105]: Yes, to keep it used.

**Review by aosewski — APPROVED** [review]

**Pull request merged**